packages/compass-generative-ai/src/components/ai-optin-modal.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+  Body,
+  Link,
+  MarketingModal,
+  SpinLoader,
+  css,
+  spacing,
+  useDarkMode,
+} from '@mongodb-js/compass-components';
+import { AISignInImageBanner } from './ai-signin-banner-image';
+import { closeOptInModal, optIn } from '../store/atlas-optin-reducer';
+import type { RootState } from '../store/atlas-ai-store';
+
+const GEN_AI_FAQ_LINK = 'https://www.mongodb.com/docs/generative-ai-faq/';
+
+type OptInModalProps = {
+  isOptInModalVisible: boolean;
+  isOptInInProgress: boolean;
+  onOptInModalClose: () => void;
+  onOptInClick: () => void;
+};
+
+const titleStyles = css({
+  marginBottom: spacing[400],
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+});
+
+const disclaimer = css({
+  padding: `0 ${spacing[900]}px`,
+});
+
+const AIOptInModal: React.FunctionComponent<OptInModalProps> = ({
```

Comment:
Let's update this modal to follow the designs a bit more. We'd like a link to the settings page, and a disabled state.
I think it's fine if we want to keep the button layout as it is here already.
Screenshots of the modals with the disabled button and the banners:
https://docs.google.com/document/d/1to8ghEzI600WxQm08T5nzNfFLkeLDx55EtYxDb4Lvno/edit?tab=t.0#bookmark=id.23kewchsjjc4

When we add that in it would probably be good to add a unit test that ensures the button is actually disabled. Here's an example unit test like that: https://github.com/mongodb-js/compass/blob/206341cf72ac9e81670400fd41235577e67c833d/packages/compass-crud/src/components/crud-toolbar.spec.tsx#L145

Reply:
changed the UI, will add in a test for the disabled button shortly